lib/stytch/b2b_sso.rb

```diff
@@ -0,0 +1,176 @@
+# frozen_string_literal: true
+
+require_relative 'request_helper'
+
+module StytchB2B
+  class SSO
+    include Stytch::RequestHelper
+    attr_reader :oidc, :saml
+
+    def initialize(connection)
+      @connection = connection
+
+      @oidc = StytchB2B::SSO::OIDC.new(@connection)
+      @saml = StytchB2B::SSO::SAML.new(@connection)
+    end
+
+    def get_connections(
+      organization_id:
+    )
+      query_params = {
+        organization_id: organization_id
+      }
+      request = request_with_query_params("/v1/b2b/sso/#{organization_id}", query_params)
+      get_request(request)
+    end
+
+    def delete_connection(
+      organization_id:,
+      connection_id:
+    )
+      delete_request("/v1/b2b/sso/#{organization_id}/connections/#{connection_id}")
+    end
+
+    def authenticate(
+      sso_token:,
+      pkce_code_verifier: nil,
+      session_token: nil,
+      session_jwt: nil,
+      session_duration_minutes: nil,
+      session_custom_claims: nil
+    )
+      request = {
+        sso_token: sso_token
+      }
+      request[:pkce_code_verifier] = pkce_code_verifier unless pkce_code_verifier.nil?
+      request[:session_token] = session_token unless session_token.nil?
+      request[:session_jwt] = session_jwt unless session_jwt.nil?
+      request[:session_duration_minutes] = session_duration_minutes unless session_duration_minutes.nil?
+      request[:session_custom_claims] = session_custom_claims unless session_custom_claims.nil?
+
+      post_request('/v1/b2b/sso/authenticate', request)
+    end
+
+    class OIDC
+      include Stytch::RequestHelper
+
+      def initialize(connection)
+        @connection = connection
+      end
+
+      def create_connection(
+        organization_id:,
+        display_name:
+      )
+        request = {
+          organization_id: organization_id,
+          display_name: display_name
+        }
+
+        post_request("/v1/b2b/sso/oidc/#{organization_id}", request)
+      end
+
+      def update_connection(
+        organization_id:,
+        connection_id:,
+        display_name:,
```

Comment:
All fields other than OrgId and ConnId are optional - we default to (and ignore) the empty string